### PR TITLE
Create board class

### LINKED ISSUE #5: Create Board class
`closes #5`

The Board class is responsible for keeping track of cells, validating coordinates, validating ship placements, placing ships, and rendering a visual representation of itself.

- Note: render method does not print put the board but returns a string to later be rendered by the Runner
render method will yield this string for a 4x4 board
`"  1 2 3 4 \nA . . . . \nB . . . . \nC . . . . \nD . . . . \n"`

- Edit this description to assign methods to each contributor

```rb
cells = {"A1"=>cell1, "A2"=>cell2, "B1"=>cell3, "B2"=>cell4}
cells.keys
=> ["A1", "A2", "B1", "B2"]
cells.values
=> [cell1, cell2, cell3, cell4]
```

### PR DESCRIPTION
**Pull request by markcyen**
closes #5

### CODE REVIEW

**Review by redferret — APPROVED** [review]
over all this was hard to do and it works. but I suggest we consider for the future to extract more methods for the logic

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      expect(board).to be_instance_of(Board)
+    end
+
+    it 'check board cells' do
+      board = Board.new
+
+      expected_cells = ["A1", "A2", "A3", "A4",
+                        "B1", "B2", "B3", "B4",
+                        "C1", "C2", "C3", "C4",
+                        "D1", "D2", "D3", "D4",]
```
Comment (redferret) [inline spec/board_spec.rb:21]: Unnecessary comma after "D4"

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  def overlap?(coords)
+    values = @cells.values
+    cells_with_ships = values.find_all do |cell|
+      cell if !cell.empty?
+    end
+
+    cells_with_ships.any? do |cell|
+      coords.include?(cell.coordinate)
+    end
+  end
```
Comment (redferret) [inline lib/board.rb:48]: Let's consider moving this into the Coordinates class. We can pass in the board's cells like so
`Coordinates.intersects?(@cells, coordinates)`
That means we will have to refactor this class, and Coordinates class as well. What do you think?

Comment (markcyen) [reply lib/board.rb:48]: I think that works. It'll be good to leverage the intersects? method in the Coordinates class to make the Board class concise. Great catch.

File: spec/board_spec.rb (hunk 1 above)
Comment (markcyen) [reply spec/board_spec.rb:21]: Ah typo. Good eye! Like your attention to details... very good strength!

**Review by markcyen — COMMENTED** [review]
Great comments and suggestions!

**Pull request merged**